servatrice/src/servatrice_database_interface.cpp

```diff
@@ -831,41 +831,43 @@ void Servatrice_DatabaseInterface::logMessage(const int senderId, const QString
     execSqlQuery(query);
 }
 
-bool Servatrice_DatabaseInterface::changeUserPassword(const QString &user, const QString &oldPassword, const QString &newPassword)
+bool Servatrice_DatabaseInterface::changeUserPassword(const QString &user, const QString &oldPassword, const QString &newPassword, const bool &force = false)
 {
     if(server->getAuthenticationMethod() != Servatrice::AuthenticationSql)
-        return true;
+        return false;
 
     if (!checkSql())
-        return true;
+        return false;
 
     QString error;
     if (!usernameIsValid(user, error))
-        return true;
+        return false;
 
-    QSqlQuery *passwordQuery = prepareQuery("select password_sha512 from {prefix}_users where name = :name");
-    passwordQuery->bindValue(":name", user);
-    if (!execSqlQuery(passwordQuery)) {
-        qDebug("Change password denied: SQL error");
-        return true;
-    }
+	QSqlQuery *passwordQuery = prepareQuery("select password_sha512 from {prefix}_users where name = :name");
```

Comment:
Can you reuse the "check password" code path from the login code rather than writing the password again here? If that's not easy, what you have here is fine for now